### PR TITLE
DrivAerML: 4L/256d — lr=5e-4 start, decay to 3e-4 at epoch 25

### PR DESCRIPTION
**Pull request by tcapelle**
## Hypothesis

Start at proven lr=5e-4 for initial convergence, then reduce to lr=3e-4 at the halfway point (~epoch 25) for fine-tuning. This two-phase LR schedule combines benefits of both LR rates — fast early convergence with fine-grained polishing in later epochs. Both lr=5e-4 and lr=3e-4 have shown value on DrivAerML; using them sequentially may capture the best of both.

## Instructions

1. Check if `train.py` supports a `--lr-decay-epoch` or `--lr-step-epoch` flag that allows stepping the LR down mid-training.
2. If such a flag exists, use it to reduce lr from 5e-4 → 3e-4 at epoch 25 with cosine T_max=30:
   ```
   cd target/icml2026 && python train.py --dataset drivaerml --optimizer adamw --lr 5e-4 --lr-decay-epoch 25 --lr-decay-factor 0.6 --cosine_t_max 30 --no-use-ema --enable-fourier --model-layers 4 --model-hidden-dim 256 --model-heads 4 --epochs 999 --batch-size 1 --drivaerml-train-surface-points 50000 --drivaerml-eval-surface-points 50000 --max-train-batches 394 --max-eval-batches 200 --wandb_group drivaerml-4L256d-lr-schedule
   ```
3. If no lr-decay flag exists, run the baseline config at lr=5e-4 and report all flags tried:
   ```
   cd target/icml2026 && python train.py --dataset drivaerml --optimizer adamw --lr 5e-4 --cosine_t_max 30 --no-use-ema --enable-fourier --model-layers 4 --model-hidden-dim 256 --model-heads 4 --epochs 999 --batch-size 1 --drivaerml-train-surface-points 50000 --drivaerml-eval-surface-points 50000 --max-train-batches 394 --max-eval-batches 200 --wandb_group drivaerml-4L256d-lr-schedule
   ```
4. Report which flags you found, the full command used, and all metrics including val_primary/surface_rel_l2_pct per epoch.

## Baseline

- **Primary metric:** `val_primary/surface_rel_l2_pct` (lower is better)
- **Current best:** 12.70% (val) / 13.54% (test) — PR #2593 (shinji)
- **Configuration:** 4L/256d, T_max=30, AdamW lr=5e-4, no-EMA, Fourier, 45 epochs
- **External target:** <3.71% (AB-UPT, ~500 epochs) — 3.4x gap remaining
- **Baseline W&B run:** 3aaevlho
- **Reproduce baseline:**
  ```
  cd target/icml2026 && python train.py --dataset drivaerml --optimizer adamw --lr 5e-4 --cosine_t_max 30 --no-use-ema --enable-fourier --model-layers 4 --model-hidden-dim 256 --model-heads 4 --epochs 999 --batch-size 1 --drivaerml-train-surface-points 50000 --drivaerml-eval-surface-points 50000 --max-train-batches 394 --max-eval-batches 200
  ```

## CRITICAL OOM Flags

Always pass these to avoid OOM:
```
--batch-size 1 --drivaerml-train-surface-points 50000 --drivaerml-eval-surface-points 50000 --max-train-batches 394 --max-eval-batches 200 --model-heads 4
```

### CODE REVIEW

**Comment by tcapelle** [comment]
ADVISOR: Closing PR #2656 because ADVISOR: Closing — DrivAerML baseline shifted to 5.73% (4L/384d, PR #2602). All 4L/256d experiments are now obsolete — the best 4L/256d result was 11.97%. Width scaling is the dominant lever. Redirecting your GPU to 4L/384d experiments. You'll receive a new assignment shortly..
